app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/home/VisitorHomeFragment.kt

```diff
@@ -41,8 +45,18 @@ class VisitorHomeFragment : Fragment() {
     ): View? {
         // Inflate the layout for this fragment
         val fragmentView = inflater.inflate(R.layout.fragment_home_visitor, container, false)
-        listUpcomingEventsLayout =
-            fragmentView.findViewById<LinearLayout>(R.id.id_upcoming_events_list)
+
+        recyclerView = fragmentView.findViewById(R.id.id_recycler_upcomming_events)
+
+        val openEvent = { event: Event ->
+            val intent = Intent(inflater.context, EventActivity::class.java).apply {
+                putExtra(EXTRA_EVENT_ID, event.eventId)
+            }
+            startActivity(intent)
+        }
+
+        recyclerView.adapter = EventItemAdapter(events, openEvent)
+        recyclerView.setHasFixedSize(false)
```

Comment:
Really nice to have removed this linear layout !

Reply:
Yes